src/main/java/teammates/ui/controller/InstructorCourseEditPageData.java

```diff
@@ -7,23 +8,119 @@
 import teammates.common.datatransfer.InstructorAttributes;
 import teammates.common.util.Const;
 import teammates.common.util.Url;
+import teammates.ui.template.CourseEditInstructorPanel;
+import teammates.ui.template.ElementTag;
 
 
 public class InstructorCourseEditPageData extends PageData {
-
+    public int offset;
+    public ElementTag deleteCourseButton;
+    public CourseAttributes course;
+    public List<CourseEditInstructorPanel> instructorPanelList;
+    public CourseEditInstructorPanel addInstructorPanel;
+    public InstructorAttributes currentInstructor;
+    public ElementTag addInstructorButton;
+    
     public InstructorCourseEditPageData(AccountAttributes account) {
         super(account);
-        index = -1;
+        instructorPanelList = new ArrayList<CourseEditInstructorPanel>();
+    }
+    
+    public void init(CourseAttributes course, List<InstructorAttributes> instructorList, InstructorAttributes currentInstructor, 
+                     int offset, List<String> sectionNames, List<String> feedbackNames) {
+        this.course = course;
+        this.offset = offset;
+        this.currentInstructor = currentInstructor;
+        
+        boolean isDisabled = !currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE);
+        deleteCourseButton = createButton("<span class=\"glyphicon glyphicon-trash\"></span>Delete", "btn btn-primary btn-xs pull-right",
+                                          "courseDeleteLink", getInstructorCourseDeleteLink(this.course.id, false), Const.Tooltips.COURSE_DELETE,
+                                          "return toggleDeleteCourseConfirmation('" + this.course.id + "');", isDisabled);
```

Comment:
`course` is equivalent to `this.course` by this point; there's no need for `this` when it is unnecessary.